wasm/command.go

```diff
@@ -28,30 +26,17 @@ const defaultCommandSize = 1024 // 1kB
 
 var buffer = make([]byte, defaultCommandSize)
 
-// NextCommand retrieves the next command from Conduit.
 func NextCommand(cmdReq *processorv1.CommandRequest) error {
-	// 2 tries, 1st try is with the current buffer size, if that's not enough,
-	// then resize the buffer and try again
-	for i := 0; i < 2; i++ {
-		// request Conduit to write the command to the given buffer
-		ptr := unsafe.Pointer(&buffer[0])
-		cmdSize := _commandRequest(ptr, uint32(cap(buffer)))
-
-		switch {
-		case cmdSize >= ErrorCodeStart: // error codes
-			return NewErrorFromCode(cmdSize)
-		case cmdSize > uint32(cap(buffer)): // not enough memory
-			buffer = make([]byte, cmdSize) // resize buffer
-			continue                       // try again
-		}
-
-		// parse the command
-		if err := proto.Unmarshal(buffer[:cmdSize], cmdReq); err != nil {
-			return fmt.Errorf("failed unmarshalling %v bytes into proto type: %w", cmdSize, err)
-		}
-		return nil
+	commandRequestCaller := HostCaller{Func: _commandRequest}
+	buf, cmdSize, err := commandRequestCaller.Call(buffer, cap(buffer))
```

Comment:
If the buffer gets resized in the host call we should overwrite `buffer` with the returned `buf` so we have the bigger allocated slice prepared for next time.